services/ticker crash on "ingest assets"

### What did you do?

<!--
If possible, provide a recipe for reproducing the error.
A complete working example or script is great.
A link on an online code editor is best. The following online editors are great.
 - JS: https://runkit.com or https://jsfiddle.net
 - Go: https://play.golang.org
 - Others: https://repl.it
-->

I was trying to ingest asset data into fresh DB like this:
```
ticker ingest assets --db-url=XXX
```

### What did you expect to see?

Successful run

### What did you see instead?

It run for around 10s and then failed:
```
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9048" pid=1
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9049" pid=1
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9050" pid=1
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9051" pid=1
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9052" pid=1
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9053" pid=1
time="2020-02-26T12:20:36.931Z" level=debug msg="Total assets processed: 9054" pid=1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x8b8c22]

goroutine 401 [running]:
github.com/stellar/go/services/ticker/internal/scraper.fetchTOMLData(0xc000650140, 0x35, 0x0, 0xc00056b530, 0x10, 0xc00056b529, 0x3, 0xc000650180, 0x38, 0xc000578ff0, ...)
        /src/ticker/services/ticker/internal/scraper/asset_scraper.go:78 +0x122
github.com/stellar/go/services/ticker/internal/scraper.processAsset(0xc000650140, 0x35, 0x0, 0xc00056b530, 0x10, 0xc00056b529, 0x3, 0xc000650180, 0x38, 0xc000578ff0, ...)
        /src/ticker/services/ticker/internal/scraper/asset_scraper.go:218 +0x1c2
github.com/stellar/go/services/ticker/internal/scraper.(*ScraperConfig).parallelProcessAssets.func1(0x13, 0x2450, 0xc00092c000, 0x2450, 0x2a49, 0xf02a01, 0xc00050e0c0, 0xc00050e0b8, 0xc0001be6c0, 0x153a)
        /src/ticker/services/ticker/internal/scraper/asset_scraper.go:256 +0x264
created by github.com/stellar/go/services/ticker/internal/scraper.(*ScraperConfig).parallelProcessAssets
        /src/ticker/services/ticker/internal/scraper/asset_scraper.go:247 +0x1e4
```

Extra info:
If I add "--testnet" flag it works fine

--- a/services/ticker/internal/scraper/asset_scraper.go
+++ b/services/ticker/internal/scraper/asset_scraper.go
@@ -1,7 +1,7 @@
 package scraper
 
 import (
-	"errors"
+	// "errors"
 	"fmt"
 	"io/ioutil"
 	"math"
@@ -17,6 +17,7 @@ import (
 	horizonclient "github.com/stellar/go/clients/horizonclient"
 	hProtocol "github.com/stellar/go/protocols/horizon"
 	"github.com/stellar/go/services/ticker/internal/utils"
+	"github.com/stellar/go/support/errors"
 )
 
 // shouldDiscardAsset maps the criteria for discarding an asset from the asset index
@@ -76,7 +77,7 @@ func fetchTOMLData(asset hProtocol.AssetStat) (data string, err error) {
 
 	req, err := http.NewRequest("GET", tomlURL, nil)
 	if err != nil {
-		err = errors.New("Invalid URL or request")
+		err = errors.Wrap(err, "invalid URL or request")
 		return
 	}
 

--- a/services/ticker/internal/scraper/asset_scraper.go
+++ b/services/ticker/internal/scraper/asset_scraper.go
@@ -69,6 +69,12 @@ func fetchTOMLData(asset hProtocol.AssetStat) (data string, err error) {
 		return
 	}
 
+	_, err = url.ParseRequestURI(tomlURL)
+	if err != nil {
+		err = errors.New("Asset has an invalid TOML URL")
+		return
+	}
+
 	timeout := time.Duration(10 * time.Second)
 	client := http.Client{
 		Timeout: timeout,

--- a/services/ticker/internal/scraper/asset_scraper.go
+++ b/services/ticker/internal/scraper/asset_scraper.go
@@ -1,7 +1,6 @@
 package scraper
 
 import (
-	// "errors"
 	"fmt"
 	"io/ioutil"
 	"math"

--- a/services/ticker/internal/scraper/asset_scraper.go
+++ b/services/ticker/internal/scraper/asset_scraper.go
@@ -69,20 +69,18 @@ func fetchTOMLData(asset hProtocol.AssetStat) (data string, err error) {
 		return
 	}
 
-	_, err = url.ParseRequestURI(tomlURL)
-	if err != nil {
-		err = errors.New("Asset has an invalid TOML URL")
-		return
-	}
-
 	timeout := time.Duration(10 * time.Second)
 	client := http.Client{
 		Timeout: timeout,
 	}
 
 	req, err := http.NewRequest("GET", tomlURL, nil)
-	req.Header.Set("User-Agent", "Stellar Ticker v1.0")
+	if err != nil {
+		err = errors.New("Invalid URL or request")
+		return
+	}
 
+	req.Header.Set("User-Agent", "Stellar Ticker v1.0")
 	resp, err := client.Do(req)
 	if err != nil {
 		return

--- a/services/ticker/internal/scraper/asset_scraper_test.go
+++ b/services/ticker/internal/scraper/asset_scraper_test.go
@@ -137,5 +137,5 @@ func TestIgnoreInvalidTOMLUrls(t *testing.T) {
 	assetStat.Links.Toml = hal.Link{Href: invalidURL}
 
 	_, err := fetchTOMLData(assetStat)
-	assert.Error(t, err)
+	assert.EqualError(t, err, "invalid URL or request: parse https:// there is something wrong here.com/stellar.toml: invalid character \" \" in host name")
 }

--- a/services/ticker/internal/scraper/asset_scraper_test.go
+++ b/services/ticker/internal/scraper/asset_scraper_test.go
@@ -5,6 +5,7 @@ import (
 	"testing"
 
 	hProtocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/go/support/render/hal"
 	"github.com/stretchr/testify/assert"
 )
 
@@ -128,4 +129,13 @@ func TestIsDomainVerified(t *testing.T) {
 	orgURL = "https://stellar.com/"
 	hasCurrency = true
 	assert.False(t, isDomainVerified(orgURL, tomlURL, hasCurrency))
+}
+
+func TestIgnoreInvalidTOMLUrls(t *testing.T) {
+	invalidURL := "https:// there is something wrong here.com/stellar.toml"
+	assetStat := hProtocol.AssetStat{}
+	assetStat.Links.Toml = hal.Link{Href: invalidURL}
+
+	_, err := fetchTOMLData(assetStat)
+	assert.Error(t, err)
 }

